annotations/builder/src/main/java/io/sundr/builder/internal/functions/ToMethod.java

```diff
@@ -318,18 +318,25 @@ private List<Statement> getStatements(Property property, List<ClassRef> alsoImpo
       }
 
       if (!descendants.isEmpty()) {
+        statements.add(new StringStatement(
+            "if (" + argumentName + "==null){ this." + fieldName + " = null; _visitables.remove(\"" + fieldName + "\");"
+                + " return (" + returnType + ") this;}"));
         for (Property descendant : descendants) {
           ClassRef dunwraped = new ClassRefBuilder(
               (ClassRef) combine(UNWRAP_COLLECTION_OF, UNWRAP_ARRAY_OF, UNWRAP_OPTIONAL_OF).apply(descendant.getTypeRef()))
                   .withArguments(new TypeRef[0]).build();
           ClassRef builderRef = BUILDER_REF.apply(dunwraped);
           statements.add(new StringStatement("if (" + argumentName + " instanceof " + dunwraped + "){ this." + fieldName
               + "= new " + builderRef + "((" + dunwraped + ")" + argumentName + "); _visitables.get(\"" + fieldName
-              + "\").add(this." + fieldName + ");}"));
-
+              + "\").clear(); _visitables.get(\"" + fieldName + "\").add(this." + fieldName + ");}"));
           alsoImport.add(dunwraped);
           alsoImport.add(builderRef);
         }
+        statements.add(new StringStatement(
+            "else {  VisitableBuilder<? extends " + ((ClassRef) type).getFullyQualifiedName()
```

Comment:
I think that we can eliminate the `else` here as the if returns.

Reply:
Removed now.